### PR TITLE
text_splitter: Added Semantic text splitter for better performance and more option of splitter to user

### PR DESCRIPTION
**Pull request by rgupta2508**
Adding 
    SemanticCharacterTextSplitter
    SemanticTiktokenTextSplitter
There was potential for better performance as well as better semantic chunking.

### CODE REVIEW

File: libs/text-splitters/langchain_text_splitters/character.py
Comment (eyurtsev) [inline libs/text-splitters/langchain_text_splitters/character.py:587]:
```suggestion
        *,
        max_characters: int = 200,
```

File: libs/text-splitters/langchain_text_splitters/character.py
Comment (eyurtsev) [inline libs/text-splitters/langchain_text_splitters/character.py:618]:
```suggestion
        self, *, max_characters: int = 200, trim_chunks: bool = False, **kwargs: Any
```

File: docs/api_reference/requirements.txt
Comment (eyurtsev) [inline docs/api_reference/requirements.txt:17]: accidental modification

File: libs/text-splitters/langchain_text_splitters/character.py
Comment (eyurtsev) [inline libs/text-splitters/langchain_text_splitters/character.py:606]: what does it mean to specify a text_splitter in the error message?

File: libs/text-splitters/pyproject.toml
Comment (eyurtsev) [inline libs/text-splitters/pyproject.toml:15]: @baskaryan what do you think optionals here?

Comment (rgupta2508) [reply libs/text-splitters/pyproject.toml]: Updated as true

File: libs/text-splitters/langchain_text_splitters/character.py
Comment (rgupta2508) [reply libs/text-splitters/langchain_text_splitters/character.py]: Updated error message

File: docs/api_reference/requirements.txt
Comment (rgupta2508) [reply docs/api_reference/requirements.txt:17]: Not changed

**Comment by rgupta2508** [comment]
One of the test which is not impacted because of this change started failing after merging from main
tests/unit_tests/agents/test_sql.py:14

**Comment by rgupta2508** [comment]
@eyurtsev please review.
Looks good now, need to fix some unrelated lint error, Finally i am able to see green light now :)

**Comment by rgupta2508** [comment]
@eyurtsev  please review.

**Comment by rgupta2508** [comment]
Removed unused, even langchain.textsplitter is also not required  as this Kept for backwards compatibility. and adding new splitter for new version.

File: libs/text-splitters/langchain_text_splitters/character.py
Comment (eyurtsev) [inline libs/text-splitters/langchain_text_splitters/character.py:583]: @rgupta2508 Could you include more context both in the PR summary and in the class documentation?

For context, what's useful to know:

1) As a maintainer, what is this package that's being wrapped? Is there a link to it so we can check whether we want to recommend it to users.
2) As a user, how is this text splitter different from other text splitters? Why would a user chose to use this over something else?

Could you include in the doc-strings a URL to the implementation?

File: libs/text-splitters/langchain_text_splitters/character.py
Comment (eyurtsev) [inline libs/text-splitters/langchain_text_splitters/character.py:589]: It doesn't look like model_name should be assigned a default value. assigning a default value will make it such that users are more likely to end up with incorrect code because they're not aware there's a configuration parameter

**Review by eyurtsev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
